lib/api/multiObjectDelete.js

```diff
@@ -233,6 +233,12 @@ function getObjMetadataAndDelete(authInfo, canonicalID, request,
                         log.debug('trying to delete locked object');
                         return callback(objectLockedError);
                     }
+                    if (versionId && objMD.location &&
+                        Array.isArray(objMD.location) && objMD.location[0]) {
+                        // we need this information for data deletes to AWS
+                        // eslint-disable-next-line no-param-reassign
+                        objMD.location[0].deleteVersion = true;
```

Comment:
nit: probably for later, but it seems strange to store this flag only in the first location, it hints that the flag should not be in the 'location' array but separate from it, somehow...